lib/validate.js

```diff
@@ -0,0 +1,90 @@
+var allowedParams = {
+	"common": ["type", "products", "currency"],
+	"impression": ["list"],
+	"click": ["list"],
+	"detail": ["list"],
+	"add": [],
+	"remove": [],
+	"purchase": ["id", "affiliation", "revenue", "tax", "shipping", "coupon"],
+	"refund": ["id", "affiliation", "revenue", "tax", "shipping", "coupon"],
+	"checkout": ["step", "option"],
+	"checkout_option": ["step", "option"]
+};
+
+var allowedProductParams = {
+	"common": ["id", "name", "brand", "category", "variant", "price"],
+	"impression": ["position"],
+	"click": ["position", "coupon"],
+	"detail": ["position", "coupon"],
+	"add": ["quantity", "coupon"],
+	"remove": ["quantity", "coupon"],
+	"purchase": ["quantity", "coupon"],
+	"refund": ["quantity", "coupon"],
+	"checkout": ["quantity", "coupon"],
+	"checkout_option": ["quantity", "coupon"]
+};
+
+function validateHit(ecData, cb) {
+	var actionCount = ecData.filter(function (dataItem) {
+		return dataItem.type !== "impression";
+	}).length;
+
+	if (actionCount > 1) {
+		cb(new TypeError("Multiple actions not allowed in one request"));
+		return;
+	}
```

Comment:
You'll thank me for this callback when we release this in production and I'll update `strict` to also accept an explicit `cb` which will allow you to pass in a void to ignore the warnings :)